### ISSUE TITLE
Error page type with 404 code can’t preview in CMS

### ISSUE THREAD

**Issue by pack254** [tractorcow-farm/silverstripe-fluent#904]
### Module version(s) affected

7.1

### Description

I have a `SilverStripe\ErrorPage\ErrorPage` page type in the project which sets the ErrorCode to 404. When I am in the CMS, to split view to preview the page. it shows the` /admin/pages` page instead. I have found this caused by [LocaleSwitchRedirector.](https://github.com/tractorcow-farm/silverstripe-fluent/blob/208b365b66005ce263dbc7ae402fe33885b26d3d/src/Middleware/LocaleSwitchRedirector.php#L19)

### How to reproduce

1. Create any two locales
2. Edit ErrorPage type with 404 code 
3. Preview the page by split view, you will see it display the admin/pages instead of error page template.

### Possible Solution

I have temporarily fixed it by using the Injector and added the additional condition for it, `!$request->getVar('CMSPreview'`).

### Additional Context

![Screenshot 2024-10-31 at 4 37 52 PM](https://github.com/user-attachments/assets/29d82793-da94-4124-8535-28bcf19d6d33)

### Validations

- [X] Check that there isn't already an issue that reports the same bug
- [X] Double check that your reproduction steps work in a fresh installation of [`silverstripe/installer`](https://github.com/silverstripe/silverstripe-installer) (with any code examples you've provided)

**Comment by tractorcow** [comment]
I think that temporary fix is fine as a permanent fix. Do you want to PR the change to add the extra condition?

**Comment by pack254** [comment]
Yep, sure thing! I will raise a [PR](https://github.com/tractorcow-farm/silverstripe-fluent/pull/905/files) for this @tractorcow 👍

**Comment by GuySartorelli** [comment]
The PR was merged
